lib/static-server.ts

```diff
@@ -374,13 +373,11 @@ export class StaticServer {
 		} else {
 			res.writeHead(status, headers);
 
-			if (req.method === 'HEAD') {
-				return this.getResult(status, headers, true);
-			}
 			try {
-				await this.stream(file, length, startByte, res);
+				if (req.method !== 'HEAD') await this.stream(file, length, startByte, res);
```

Comment:
This would still cause HEAD responses to hang. Please check my other comment/suggestion.

Reply:
Ah, because of the lack of `req.end()`; I see.